### ISSUE TITLE
Event company wise

### ISSUE THREAD

**Issue by deva199** [iomad/iomad#1256]
Hi there,
Does managing events can be set to work at company level ?
when I create any event is actually displayed to all company admins/ manager/teacher
How to overcome this scenario ?

**Comment by turf212** [comment]
Events are site wide in Moodle.  If you are trapping or reporting on events then you need to code the trap/report to filter out the companies.

If you have a shared course, the way this is handled is by using company groups and having forced separate groups on in the course.  If you elevate someones access to teacher then. by default, they can see all of the course groups/ everyone in the course.  There is an Iomad role (company course access) which is similar to the non-editing teacher role which is given to company managers when the course is non-licensed (or to any user if you have unticked the option to have managers as students in the local iomad settings and assigned them the company reporter role).

TLDR : If you are playing around with shared courses and you give someone more access than they should have, they will see outside of their company box.
